wandb/proto/wandb_internal.proto

```diff
@@ -1167,3 +1145,40 @@ message PythonPackagesRequest {
   }
   repeated PythonPackage package = 1;
 }
+
+// Path within nested configuration object.
+//
+// The path is a list of strings, each string is a key in a nested configuration
+// dict. These paths are used to filter subtrees in and out of the config
+// before we capture a schema. This gives users the ability to limit which
+// parts of the config are exposed as inputs to a job.
+message JobInputPath {
+  repeated string path = 1;
+}
+
+// Specifies a new source for job inputs.
+//
+// The source is either the run config (wandb.config) or a config file.
+// If a config file is specified, the file path is relative to
+// <run-files-dir>/configs.
+message JobInputSource {
+  enum SourceType {
+    RUN = 0;
+    FILE = 1;
+  }
+  SourceType type = 1;
+  string file_path = 2;
```

Comment:
what if you make the source a oneof with each type having it's own metadata?

Reply:
how about
```
message JobInputSource {
  oneof source {
    string file_path = 1;
    bool run_config = 2;
  }
}
```

Reply:
i was thinking more like:
message JobInputSource {
 message RunConfig{
 ....
 }
 Message FileConfig {
...
}
  oneof source {
    RunConfig run_config = 1;
    FileConfig file_config = 2;
  }
}

just to allow adding more unique info for each source, but i'm not sure if it is required, you would know better

Reply:
also making source you will only have one source at a time? oneof makes it mutex...

Reply:
That does seem easier to add onto if we end up wanting to pass more information about the source. I'm on board

Reply:
Done, this was a fantastic suggestion. The code thanks you